docs/docs/fargate-tutorials/prisma.md

```diff
@@ -134,126 +140,197 @@ node index.js
+  const repoUri = (await prisma.ecs_simplified.findFirst({
+    where: { app_name: appName },
     select: { repository_uri: true }
   })).repository_uri;
 
-}
-```
-
-The last part of the script will apply the changes described in the hosted db to your cloud account which will take a few minutes waiting for AWS
-
-```js title="my_project/migrations/index.js"
-const apply = await prisma.$queryRaw`SELECT * from iasql_apply();`
-console.dir(apply)
-```
-
-If the function call is successful, it will return a virtual table with a record for each cloud resource that has been created, deleted or updated.
-
-```sql
- action |    table_name       |   id   |      description      
---------+---------------------+--------+-----------------------
- create | public_repository   |      2 | quickstart-repository
- create | cluster             |      2 | 2
- create | task_definition     |      2 | 2
- create | service             |      2 | 2
- create | listener            |      2 | 2
- create | load_balancer       |      2 | 2
- create | target_group        |      2 | 2
- create | security_group      |      5 | 5
- create | security_group_rule |      3 | 3
- create | security_group_rule |      4 | 4
- create | role                |        | ecsTaskExecRole
-```
-
-## Login, build and push your code to the container registry
-
-1. Grab your new `ECR URI` from the hosted DB
-```bash
-QUICKSTART_ECR_URI=$(psql -At 'postgres://d0va6ywg:nfdDh#EP4CyzveFr@db.iasql.com/_4b2bb09a59a411e4' -c "
-SELECT repository_uri
-FROM repository
-WHERE repository_name = '<project-name>-repository';")
-```
-
-2. Login to AWS ECR using the AWS CLI. Run the following command and use the correct `<ECR-URI>` and AWS `<profile>`
-
-```bash
-aws ecr get-login-password --region ${AWS_REGION} --profile <profile> | docker login --username AWS --password-stdin ${QUICKSTART_ECR_URI}
-```
-
-:::caution
-
-Make sure the [CLI is configured with the same credentials](https://docs.aws.amazon.com/cli/latest/userguide/cli-configure-files.html), via environment variables or `~/.aws/credentials`, as the ones provided to IaSQL or this will fail.
-
-:::
-
-3. Build your image locally
+  const cbData = {
+    role_name: cbRole,
+    assume_role_policy_document: assumeServicePolicy,
+    attached_policies_arns: [codebuildPolicyArn, cloudwatchLogsArn, pushEcrPolicyArn]
+  }
+  await prisma.role.upsert({
+    where: { role_name: cbRole },
+    create: cbData,
+    update: cbData,
+  });
 
-```bash
-docker build -t <project-name>-repository app
-```
+  const buildSpecRes = await prisma.$queryRaw`SELECT generate_put_ecr_image_build_spec('${region}', 'latest', '${appName}-repository', '${repoUri}', 'examples/ecs-fargate/prisma/app')`;
+  const buildSpec = buildSpecRes[0]['generate_put_ecr_image_build_spec'];
 
-4. Tag your image
+  const pjData = {
+    project_name: appName,
+    source_type: 'GITHUB',
+    service_role_name: cbRole,
+    source_location: ghUrl,
+    build_spec: buildSpec,
```

Comment:
Does prisma not allow an `upsert` column value to come from a Postgres function? That's a weird limitation. Let me look into it.